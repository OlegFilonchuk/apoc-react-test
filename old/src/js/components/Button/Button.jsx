import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { upperFirst } from 'lodash';
import { getDataTestElementProps, dataTestElementPropTypes } from '../../utils/dataTestElementPropUtils';

import './Button.less';

const noop = () => null;

function Button(props) {
  const { className, isSelected, isLinkView, disabled, id, name, type, children, onClick, title, value } = props;

  const buttonClass = classNames('sc-btn', className, {
    'sc-on': isSelected,
    'sc-disabled': disabled,
    'link-view': isLinkView
  });

  const clickHandler = disabled ? noop : onClick;

  const processDataTestElement = dataTestElementValue =>
    `${dataTestElementValue}${value && typeof value === 'string' ? '-' : ''}${
      value && typeof value === 'string' ? upperFirst(value) : ''
    }`;

  return (
    <button
      id={id}
      name={name}
      type={type}
      disabled={disabled}
      className={buttonClass}
      onClick={clickHandler}
      title={title}
      {...getDataTestElementProps(props, processDataTestElement)}
    >
      {children}
    </button>
  );
}

Button.propTypes = {
  /**
   * Button id
   */
  id: PropTypes.string,

  /**
   * Is button disabled
   */
  disabled: PropTypes.bool,

  /**
   * Button name
   */
  name: PropTypes.string,

  /**
   * Button type
   */
  type: PropTypes.string,

  /**
   * Is button selected
   */
  isSelected: PropTypes.bool,

  /**
   * Is button looks like link
   */
  isLinkView: PropTypes.bool,

  /**
   * Is called when button is clicked
   */
  onClick: PropTypes.func.isRequired,

  /**
   * Inherited classes. For more details check Buttons section in Chameleon.
   * Defaults to 'sc-btn-default'.
   */
  className: PropTypes.string,

  /**
   * Button content
   */
  children: PropTypes.node.isRequired,

  /* eslint-disable react/no-unused-prop-types */
  /**
   * Button value that is used in ButtonGroup component
   */
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.object]),
  /* eslint-enable react/no-unused-prop-types */

  title: PropTypes.string,

  ...dataTestElementPropTypes
};

Button.defaultProps = {
  id: '',
  name: '',
  type: 'button',
  disabled: false,
  isSelected: false,
  isLinkView: false,
  value: null,
  className: 'sc-btn-default',
  'data-test-element': '',
  title: ''
};

export default Button;
