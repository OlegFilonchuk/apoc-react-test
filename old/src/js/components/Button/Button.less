@import '~chameleon/less/variables.less';

.sc-btn {
  cursor: pointer;

  &.link-view {
    color: @primary;
    background: none;
    padding: initial;

    &.sc-disabled {
      color: @disable;
    }
  }
}
