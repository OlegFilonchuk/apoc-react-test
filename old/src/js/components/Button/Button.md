## Basic button

Simple usage of the **Button** component:

    const callbackFn = () => {
        document.getElementById('clicked').textContent = 'I was clicked!';
    };

    (
      <div>
        <Button className="sc-btn-default" onClick={ callbackFn }>Push Me</Button>
        <div id="clicked">
            Please click button!
        </div>
      </div>
    )

## Disabled button

Rendering Button with a disables state

    <Button onClick={() => true} disabled={true}>You can't click me</Button>

## Colors

Button component can support different colors:

    const onClick = (text) => {
      console.log(`${text} clicked!`);
    };

    (
      <div>
        <p>
          <Button onClick={() => { onClick('Button normal'); }}>Button Normal</Button>
        </p>
        <p>
          <Button onClick={() => { onClick('Button Primary'); }}  className="sc-btn-primary">Button Primary</Button>
        </p>
        <p>
          <Button onClick={() => { onClick('Primary Outline'); }}  className="sc-btn-primary-outline">Button Primary Outline</Button>
        </p>
        <p>
          <Button onClick={() => { onClick('Button Warning'); }}  className="sc-btn-warning">Button Warning</Button>
        </p>
        <p>
          <Button onClick={() => { onClick('Button Error'); }}  className="sc-btn-error">Button Error</Button>
        </p>
        <p>
          <Button onClick={() => { onClick('Error Outline'); }}  className="sc-btn-error-outline">Button Error Outline</Button>
        </p>
        <p>
          <Button onClick={() => { onClick('Button Success'); }}  className="sc-btn-success">
            <i className="sc-icon-download" /> Button Success
          </Button>
        </p>
        <p>
          <Button isLinkView onClick={() => { onClick('Link Button'); }}  className="sc-primary">
            <i className="sc-icon-file" /> Link Button
          </Button>
        </p>
      </div>
    )
