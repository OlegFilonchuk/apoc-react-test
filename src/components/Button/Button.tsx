import React from 'react';

export type ButtonProps = {
  label: string;
};

export function Button(props: ButtonProps) {
  return <button>{props.label}</button>;
}
