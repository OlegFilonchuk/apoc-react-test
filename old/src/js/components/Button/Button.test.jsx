import React from 'react';
import { mount } from 'enzyme';

import Button from '../Button/Button';

const defaultProps = {
  onClick: () => null
};

describe('<Button />', () => {
  const mountButtonGroup = (children, props = {}, mountFn = mount) =>
    mountFn(
      <Button {...defaultProps} {...props}>
        {children}
      </Button>
    );

  it('should render Button component', () => {
    const button = mountButtonGroup('Foo Bar Button');

    expect(button).toMatchSnapshot();
  });

  it('should render disabled button ', () => {
    const buttonGroup = mountButtonGroup("I'm disabled button", {
      disabled: true
    });

    expect(buttonGroup).toMatchSnapshot();
  });

  it('should allow to click on enabled button ', () => {
    const fakeClick = jest.fn();

    const buttonGroup = mountButtonGroup('Click me!', {
      disabled: false,
      onClick: fakeClick
    });

    buttonGroup.find('button').simulate('click');

    expect(fakeClick).toHaveBeenCalled();
  });

  it('should not allow to click on disabled button ', () => {
    const fakeClick = jest.fn();

    const buttonGroup = mountButtonGroup("Can't touch this!", {
      disabled: true,
      onClick: fakeClick
    });

    buttonGroup.find('button').simulate('click');

    expect(fakeClick).not.toHaveBeenCalled();
  });
});
