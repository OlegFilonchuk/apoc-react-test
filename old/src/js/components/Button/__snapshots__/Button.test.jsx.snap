// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render Button component 1`] = `
<Button
  className="sc-btn-default"
  data-test-element=""
  disabled={false}
  id=""
  isLinkView={false}
  isSelected={false}
  name=""
  onClick={[Function]}
  title=""
  type="button"
  value={null}
>
  <button
    className="sc-btn sc-btn-default"
    data-test-element="undefined"
    disabled={false}
    id=""
    name=""
    onClick={[Function]}
    title=""
    type="button"
  >
    Foo Bar Button
  </button>
</Button>
`;

exports[`<Button /> should render disabled button  1`] = `
<Button
  className="sc-btn-default"
  data-test-element=""
  disabled={true}
  id=""
  isLinkView={false}
  isSelected={false}
  name=""
  onClick={[Function]}
  title=""
  type="button"
  value={null}
>
  <button
    className="sc-btn sc-btn-default sc-disabled"
    data-test-element="undefined"
    disabled={true}
    id=""
    name=""
    onClick={[Function]}
    title=""
    type="button"
  >
    I'm disabled button
  </button>
</Button>
`;
